### ISSUE TITLE
Word Obfuscation of other Profile is applied to current Profile

### ISSUE THREAD

**Issue by SVNKoch** [phw198/OutlookGoogleCalendarSync#1499]
**OGCS Version**: v2.9.5.1
**Edition**: Installed
**Sync Direction**: G->O

**Bug Description**  
Setup:
I have many profiles each for one of my google calendars: "Private" "Travel time blocker" "University" ...
I use word obfuscation to rename these entries to the name of the calendar using the Regex "^.*$". So in the "Private" Profile every item should be named "Private" and so one for all other calendars as well.
This works perfectly when doing **full** syncs.

Bug:
Every few days (when I'm online) I have calendar entries which are synced from the "University" calendar named "Private" or "Travel time blocker" named "University". Because of that I then manually do a full Sync for each profile which then fixes the naming issues again. The normal sync ignores the wrong name.

These profiles however also apply a category (like "Private" & "University" -> Private, "Travel time blocker" -> Blocker). These categories are correct only the name has issues.

This makes me thing the wrong obfuscation rule is applied.

Reading the logs ([shortenedLog.log](https://github.com/phw198/OutlookGoogleCalendarSync/files/8928669/shortenedLog.log)) there are these lines:
```
2022-06-17 16:43:38,317 INFO  1 OutlookGoogleCalendarSync.SettingsStore.Calendar [279] -    Obfuscate Words: True
2022-06-17 16:43:38,318 INFO  1 OutlookGoogleCalendarSync.SettingsStore.Calendar [283] -      '^.*$' -> 'Driving'
```
and
```
2022-06-17 16:43:48,988 DEBUG  20 OutlookGoogleCalendarSync.OutlookOgcs.Calendar [446] -  Processing >> 17.06.2022 17:30 => "Driving to event"
2022-06-17 16:43:49,034 DEBUG  20 OutlookGoogleCalendarSync.Console [285] -  17.06.2022 17:30 "Driving to event"
2022-06-17 16:43:49,035 DEBUG  20 OutlookGoogleCalendarSync.Console [285] -  AttributeChangeStart time17.06.2022 17:45:00 → 17.06.2022 17:30:00SubjectDriving → University
```

the correct word obfuscation rule is loaded (=> "Driving") but then the subject is replaced with "University" instead.

I also attached a very short excerpt of my setings file showing the obfuscation profile rules.
[important settings Excerpt.txt](https://github.com/phw198/OutlookGoogleCalendarSync/files/8928675/important.settings.Excerpt.txt)

**Comment by SVNKoch** [comment]
@phw198 can you reopen this one?

**Comment by phw198** [comment]
:package: Thanks for the detailed issue report - it should now be resolved with [v2.9.5.9.zip](https://github.com/phw198/OutlookGoogleCalendarSync/files/9334196/v2.9.5.9.zip) ([see instructions for applying](https://github.com/phw198/OutlookGoogleCalendarSync/wiki/Reporting-Problems#applying-a-hotfix)).

Please let me know how it goes.

**Comment by SVNKoch** [comment]
Thanks for fixing it . I applied the patch before my vacation and since then I haven't had wrongly named calendar instances. 👍
